### PR TITLE
666 support knime highlighting feature

### PR DESCRIPTION
**Pull request by ahmadswaid**

### CODE REVIEW

**Review by miguelalba — CHANGES_REQUESTED** [review]

File: de.bund.bfr.knime.fsklab.nodes/js-src/de/bund/bfr/knime/fsklab/v1.9/joiner/joiner.js [hunk 1]
```diff
@@ -31,11 +31,32 @@ joiner = function () {
...
   window.joinRelationsMap = {};
-
+  let poolSize = 0;
+  var selectionChanged = function (modelMetaData) {
+    console.log('in joiner', modelMetaData);
+    //extractAndCreateUI(JSON.stringify(modelMetaData.changeSet.added[0]));
+
+    if (poolSize == 0) {
+      let selectedModel = modelMetaData.changeSet.added[0][0];
+      editModelsPool('firstModel', selectedModel.modelMath.parameter, selectedModel.generalInformation.name, selectedModel, selectedModel.modelType);
```
[the remark this answers is not in the document]
Comment (ahmadswaid) [reply de.bund.bfr.knime.fsklab.nodes/js-src/de/bund/bfr/knime/fsklab/v1.9/joiner/joiner.js]: The function is called in other places with arguments of other Object which has another structure.
please look to the call of it with arguments from _joinerModelsData Object

**Comment by ahmadswaid** [comment]
@miguelalba Unfortunately, I thought if push another commit it will update the PR like with MR in Gitlab.
so I don't know if you noticed that files were labeled with outdated.
I did all the requested changes but don't know if I have to create another PR to have the branch eligible for merging.

File: de.bund.bfr.knime.fsklab.nodes/js-src/de/bund/bfr/knime/fsklab/v1.9/joiner/joiner.js [hunk 2]
```diff
@@ -580,9 +726,9 @@ joiner = function () {
...
       },
-      'markup': `<circle fill="#FF7979"  r="92.5" class="port-body"><title>Parameter ID: ` + param.id + `\nParameter Name: ` + param.name + `\nDescription: ` + param.description + `\nUnit: ` + param.unit + `\nDataType: ` + param.dataType + `\nSource: ` + param.source + `\nSubject: ` + param.subject + `\nDistribution: ` + param.distribution + `\nReference: ` + param.reference + `\nVariabilitySubject: ` + param.variabilitySubject + `\nMinValue: ` + param.minValue + `\nMaxValue: ` + param.maxValue + `\nError: ` + param.error + `\n</title></circle>`,
+      'markup': `<circle fill="#FF7979"  r="92.5" class="port-body"><title>Parameter ID: ` + param.idmask + `\nParameter Name: ` + param.name + `\nDescription: ` + param.description + `\nUnit: ` + param.unit + `\nDataType: ` + param.dataType + `\nSource: ` + param.source + `\nSubject: ` + param.subject + `\nDistribution: ` + param.distribution + `\nReference: ` + param.reference + `\nVariabilitySubject: ` + param.variabilitySubject + `\nMinValue: ` + param.minValue + `\nMaxValue: ` + param.maxValue + `\nError: ` + param.error + `\n</title></circle>`,
```
Comment (miguelalba) [inline de.bund.bfr.knime.fsklab.nodes/js-src/de/bund/bfr/knime/fsklab/v1.9/joiner/joiner.js:731]: We could use a template string here.

**Review by miguelalba — APPROVED** [review]
It is fine. I would use a template string for the markup in joiner.js:731 but I do not to slow this PR more.

**Pull request merged**